braid/_braid.c

```diff
@@ -1428,6 +1429,18 @@ _braid_Step(braid_Core         core,
       rfactors[ii] = _braid_StatusElt(status, rfactor);
       if ( !_braid_CoreElt(core, r_space) && _braid_StatusElt(status, r_space) )
             _braid_CoreElt(core, r_space) = 1;
+
+      /* If NOT called from within FRefine, store the user's dtvalues, if set */
+      braid_Int caller = _braid_CoreElt(core, calling_function);
+      if (caller != braid_ASCaller_FRefine)  
```

Comment:
@steffi7574 This is the only place we use the ASCaller stuff in braid logic (it appears).  Can we do this differently?  Does it really matter where the routine was called?

Reply:
This was overly complicated, I used it for preventing memory leaks when FRefine call's _braid_Step again. I changed the memory handling, and removed the ASCaller_FRefine restriction.  Now rdtvalues are free'd at the end of FRefine. I added an option to ex-01-refinement to test this feature.